samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -109,3 +108,43 @@ def _get_configuration_address(address: str) -> str:
         The address clean of indices
     """
     return re.sub(r"\[[^\[\]]*\]", "", address)
+
+
+def _resolve_module_output(module, output_name):
+    pass
+
+
+def _resolve_module_variable(module: TFModule, variable_name: str) -> List[Union[ConstantValue, ResolvedReference]]:
+    # return a list of the values that resolve the passed variable
+    # name in the input module.
+    results: List[Union[ConstantValue, ResolvedReference]] = []
+
+    var_value = module.variables[variable_name]
+
+    # check the possible constant value for this variable
+    if isinstance(var_value, ConstantValue) and var_value is not None:
+        results.append(ConstantValue(var_value.value))
+
+    # check the possible references value for this variable
+    if isinstance(var_value, References) and var_value is not None:
+        cleaned_references = _clean_references_list(var_value.value)
+        for reference in cleaned_references:
+            # refer to a variable passed to this module from its parent module
+            if reference.startswith("var."):
+                config_var_name = _get_configuration_address(reference[4:])
+                if module.parent_module:
+                    results += _resolve_module_variable(module.parent_module, config_var_name)
+            # refer to another module output. This module will be defined in the same level as this module
+            elif reference.startswith("module."):
+                module_name = reference[reference.find(".") + 1 : reference.rfind(".")]
+                config_module_name = _get_configuration_address(module_name)
+                output_name = reference[reference.rfind(".") + 1 :]
+                if module.parent_module and module.parent_module.child_modules:
+                    results += _resolve_module_output(
+                        module.parent_module.child_modules[config_module_name], output_name
+                    )
+            # this means either a resource, data source, or local.variables.
+            elif module.full_address is not None:
+                results.append(ResolvedReference(reference, module.full_address))
```

Comment:
I think we should return the parent module address instead of the current module address. If  the value of the variable is not a constant value, so it means it got passed from the parent module.